### PR TITLE
Grouped Mode

### PR DESCRIPTION
**Pull request by masterrr**
Closing
https://github.com/toggl/mobile/issues/795
https://github.com/toggl/mobile/issues/811
https://github.com/toggl/mobile/issues/812
https://github.com/toggl/mobile/issues/827
https://github.com/toggl/mobile/issues/828

### CODE REVIEW

File: Ross/ViewControllers/LogViewController.cs [hunk 1]
```diff
@@ -75,16 +99,27 @@ public override void ViewDidLoad ()
...
                 headerView.AdaptToTableView (TableView);
+
+                var bus = ServiceContainer.Resolve<MessageBus> ();
+                subscriptionSettingChanged = bus.Subscribe<SettingChangedMessage> (OnSettingChanged);
+            }
+
+            private void OnSettingChanged (SettingChangedMessage msg)
+            {
+                // Protect against Java side being GCed
+                if (Handle == IntPtr.Zero) {
```
Comment (monday8am) [inline Ross/ViewControllers/LogViewController.cs]: Java code in iOS :P

File: Ross/ViewControllers/LogViewController.cs [hunk 2]
```diff
@@ -315,12 +426,12 @@ private void OnTagsUpdated (object sender, EventArgs args)
                 RebindTags ();
```
Comment (monday8am) [inline Ross/ViewControllers/LogViewController.cs]: We have lost the tag view. Check the TimeEntryHolder NumberOfTags property ..
https://github.com/toggl/mobile/blob/master/Phoebe/Data/Utils/TimeEntryHolder.cs#L92

File: Ross/ViewControllers/LogViewController.cs [hunk 3]
```diff
@@ -126,23 +174,60 @@ public override void Attach ()
...
-            }
+                // this line is needed to update
+                // the section list every time
+                // the collection is updated.
+                base.OnCollectionChange (sender, e);
 
-            protected override IEnumerable<AllTimeEntriesView.DateGroup> GetSections ()
-            {
-                return dataView.DateGroups;
+                TableView.BeginUpdates ();
```
Comment (monday8am) [inline Ross/ViewControllers/LogViewController.cs]: @masterrr Implement a condition for Move events, another for Reset (ReloadData) and we could improve the Add condition to support AddRange: we could detect a long list of rows and add them in one time. This last is not needed but we'll give us better performance.

File: Ross/ViewControllers/LogViewController.cs [hunk 4]
```diff
@@ -59,11 +59,35 @@ protected override void Dispose (bool disposing)
...
 
+            private void EnsureAdapter (bool forceRebind = false)
+            {
+                if (TableView.Source == null || forceRebind) {
+                    TableView.Hidden = true;
+                    if (TableView.Source != null) {
+                        (TableView.Source as Source).Dispose ();
+                    }
+                    var isGrouped = ServiceContainer.Resolve<ISettingsStore> ().GroupedTimeEntries;
+                    var collectionView = isGrouped ? (TimeEntriesCollectionView)new GroupedTimeEntriesView () : new LogTimeEntriesView ();
```
Comment (monday8am) [inline Ross/ViewControllers/LogViewController.cs]: @masterrr Be sure to dispose old CollectionView objects when you change to a new one. Dispose it if needed in the Dispose () method of class.

**Comment by taavirehemagi** [comment]
What's the status of this? Is anyone going to do something about this?

**Comment by masterrr** [comment]
I can resolve conflicts w/ master if needed

**Comment by monday8am** [comment]
.. not for the moment but who knows ;) thanks anyway!

**Comment by monday8am** [comment]
out of date.

**Pull request closed without merging**